packages/linux-firmware/linux-firmware.spec

```diff
@@ -0,0 +1,89 @@
+%global debug_package %{nil}
+
+%global _firmwarepath   /usr/lib/firmware
+
+# Many of the firmware files have specialized binary formats that are not supported
+# by the strip binary used in __spec_install_post macro. Work around build failures
+# by skipping striping.
+%global __strip /usr/bin/true
+
+Name:       %{_cross_os}linux-firmware
+Version:    20230625
+Release:    1%{?dist}
+Summary:    Firmware files used by the Linux kernel
+License:    GPL+ and GPLv2+ and MIT and Redistributable, no modification permitted
```

Comment:
So if I am reading the SPDX grammar and License list correctly this would be something like this:

```
GPL-1.0-or-later AND GPL-2.0-or-later AND MIT AND ?
```

A lot of the firmware is licensed as "Redistributable, no modification permitted" (or that is at least the name for that bucket that fedora uses. All the firmware we currently have left to include falls into that group. If I understand correctly there is currently no solution for these in the SPDX guidelines. The [Fedora licensing guide](https://docs.fedoraproject.org/en-US/legal/update-existing-packages/#_redistributable_no_modification_permitted) marks usage of that with SPDX as unknown.

What do can we make of this?

Reply:
So I looked at the `LICEN[CS]E` files we ship with what is proposed here. That is the following list:

```
GPL-2 - clear what to do
GPL-3 - clear what to do
LICENCE.chelsio_firmware - LicenseRef-scancode-chelsio-linux-firmware 
LICENCE.myri10ge_firmware - seems to be no available
LICENCE.phanfw - LicenseRef-scancode-qlogic-firmware is identical
LICENSE.i915 - seems to be not available and not matching any other intel license
LICENSE.ice - seems to be not available and not matching any other intel license
LICENSE.ice_enhanced - seems to be not available and not matching any other intel license
```

Refering to the lists of Licenses in https://spdx.org/licenses/ and https://scancode-licensedb.aboutcode.org/

Reply:
After reaching out to ospo and playing around some more with the scancode tooling I have come up with a list of SPDX license reference codes. I think we are fine with those, although I am not super happy about `LicenseRef-scancode-free-unknown` for the license of `tg3` as specified in `WHENCE`.